client/signup/steps/site-info-collection/index.tsx

```diff
@@ -0,0 +1,277 @@
+		siteInfoCollectionDataFromStore
+	);
+	const onChangeField = ( { target: { name, value } }: ChangeEvent< HTMLInputElement > ) => {
+		setFormValues( {
+			...formValues,
+			[ name ]: value,
+		} );
+	};
+
+	const [ formErrors, setFormErrors ] = useState< ValidationErrors >( {} );
+
+	const sections: SectionProps[] = [
+		{
+			title: translate( '%d. Name of your business', {
+				args: [ 1 ],
+				comment: 'This is the serial number: 1',
+			} ),
+			component: (
+				<TextInputField
+					label={ translate(
+						'Please enter the name of your business as you want it to appear on your new site.'
+					) }
+					value={ formValues.siteTitle }
+					error={ formErrors.siteTitle }
+					name="siteTitle"
+					onChange={ onChangeField }
+				/>
+			),
+			showSkip: false,
+			summary: formValues.siteTitle,
+			validate: ( { siteTitle } ) => {
+				const isValid = Boolean( siteTitle && siteTitle.length );
+				return {
+					result: isValid,
+					errors: {
+						siteTitle: isValid ? null : translate( 'Please enter a valid site title.' ),
+					},
+				};
+			},
+		},
+		{
+			title: translate( '%d. Site Description', {
+				args: [ 2 ],
+				comment: 'This is the serial number: 2',
+			} ),
+			component: (
+				<TextInputField
+					label={ translate( 'A short description for your website.' ) }
+					value={ formValues.siteDescription }
+					name="siteDescription"
+					onChange={ onChangeField }
+				/>
+			),
+			showSkip: true,
+			summary: formValues.siteDescription,
+		},
+		{
+			title: translate( '%d. Social Media Profiles', {
+				args: [ 3 ],
+				comment: 'This is the serial number: 3',
+			} ),
+			component: (
+				<SocialMediaProfiles
+					facebookProps={ {
+						value: formValues.facebookUrl,
+						name: 'facebookUrl',
+					} }
+					twitterProps={ {
+						value: formValues.twitterUrl,
+						name: 'twitterUrl',
+					} }
+					linkedinProps={ {
+						value: formValues.linkedinUrl,
+						name: 'linkedinUrl',
+					} }
+					instagramProps={ {
+						value: formValues.instagramUrl,
+						name: 'instagramUrl',
+					} }
+					onChange={ onChangeField }
+				/>
+			),
+			showSkip: true,
+			summary: undefined,
+		},
+		{
+			title: translate( '%d. Contact Information', {
+				args: [ 4 ],
+				comment: 'This is the serial number: 4',
+			} ),
+			component: (
+				<ContactInformation
+					displayEmailProps={ {
+						value: formValues.displayEmail,
+						name: 'displayEmail',
+					} }
+					displayPhoneProps={ {
+						value: formValues.displayPhone,
+						name: 'displayPhone',
+					} }
+					displayAddressProps={ {
+						value: formValues.displayAddress,
+						name: 'displayAddress',
+					} }
+					onChange={ onChangeField }
+				/>
+			),
+			showSkip: true,
+			summary: undefined,
+		},
+	];
+
+	const isSectionAtIndexTouchedInitialState: Record< string, boolean > = {};
+	for ( let i = 0; i <= currentIndex; i++ ) {
+		isSectionAtIndexTouchedInitialState[ `${ i }` ] = true;
```

Comment:
Sorry for not putting in a comment here! The idea of this function is to initialize the touched states upto the value of the `currentIndex` from redux. I have assumed that if the user is on a specific section and they refresh the page, we can mark all sections as touched up to that current index. I know this is in direct conflict with my comment [here](https://github.com/Automattic/wp-calypso/pull/59986#discussion_r784814708), but I could not think of any other solution other than storing the touched states in redux. IMO, storing touched states in redux seems like overkill. 🙂

Reply:
Actually I think since this is going to be an extremely long form to be filled, overkill UX solution like saving touched states in redux is exactly what we need 😁 .

But we might have to expand on this paradigm later. For example I think what we need to maintain is whether the user "completed" the value not whether they just merely "touched". And we should solve the complexity of how validation ties with it (We should save the invalid value in redux but mark the field as "invalid". Then we have to think about showing an invalid label when the form field is contracted.

This is where we innovate for DIFM IMHO. Radically optimising the long form filling process.

But yes these can all be future enhancement. Kudos for thinking of this early on 💯 🥇 👏 . This did not cross my mind 😬.